launcher/Application.cpp

```diff
@@ -285,12 +285,29 @@ Application::Application(int &argc, char **argv) : QApplication(argc, argv)
             return;
         }
     }
+
     m_instanceIdToLaunch = args["launch"].toString();
     m_serverToJoin = args["server"].toString();
     m_profileToUse = args["profile"].toString();
     m_liveCheck = args["alive"].toBool();
     m_zipToImport = args["import"].toUrl();
 
+    // error if --launch is missing with --server
+    if(m_instanceIdToLaunch.isEmpty() && !m_serverToJoin.isEmpty())
```

Comment:
why can't you combine these two

Reply:
We could. I just wanted to keep the original error messages.

Reply:
I see no problem with combining them in that case. We're finally completely diverting from MultiMC and there's currently no need to keep its standards, messages, etc.

except for cat